metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -493,25 +502,60 @@ def _parse_exported_bigquery_audit_metadata(
 
     def _create_lineage_map(self, entries: Iterable[QueryEvent]) -> Dict[str, Set[str]]:
         lineage_map: Dict[str, Set[str]] = collections.defaultdict(set)
-        num_entries: int = 0
-        num_skipped_entries: int = 0
+        self.report.num_total_lineage_entries = 0
+        self.report.num_skipped_lineage_entries_missing_data = 0
+        self.report.num_skipped_lineage_entries_not_allowed = 0
+        self.report.num_skipped_lineage_entries_other = 0
         for e in entries:
-            num_entries += 1
-            if e.destinationTable is None or not e.referencedTables:
-                num_skipped_entries += 1
+            self.report.num_total_lineage_entries += 1
+            if e.destinationTable is None or not (
+                e.referencedTables or e.referencedViews
+            ):
+                self.report.num_skipped_lineage_entries_missing_data += 1
+                continue
+            # Skip if schema/table pattern don't allow the destination table
+            destination_table_str = str(e.destinationTable.remove_extras())
+            destination_table_str_parts = destination_table_str.split("/")
+            if not self.config.schema_pattern.allowed(
+                destination_table_str_parts[3]
+            ) or not self.config.table_pattern.allowed(destination_table_str_parts[-1]):
+                self.report.num_skipped_lineage_entries_not_allowed += 1
                 continue
+
             entry_consumed: bool = False
+            has_table = False
             for ref_table in e.referencedTables:
-                destination_table_str = str(e.destinationTable.remove_extras())
                 ref_table_str = str(ref_table.remove_extras())
                 if ref_table_str != destination_table_str:
                     lineage_map[destination_table_str].add(ref_table_str)
                     entry_consumed = True
+                    has_table = True
+            has_view = False
+            for ref_view in e.referencedViews:
+                ref_view_str = str(ref_view.remove_extras())
+                if ref_view_str != destination_table_str:
+                    lineage_map[destination_table_str].add(ref_view_str)
+                    entry_consumed = True
+                    has_view = True
+            if has_table and has_view:
+                # TODO: More comments required on why this is needed.
```

Comment:
@anshbansal , please address the TODO. Also, it looks like we don't need `entry_consumed` as it can be replaced with `(has_table or has_view)`. Please remove it.